app/src/main/java/com/amaze/filemanager/fragments/FtpServerFragment.java

```diff
@@ -656,4 +666,38 @@ private void setSecurePreference(boolean isSecureEnabled) {
         .putBoolean(FtpService.KEY_PREFERENCE_SECURE, isSecureEnabled)
         .apply();
   }
+
+  private void promptUserToEnableWireless(@Nullable NetworkInfo ni) {
+    // No wifi, no data, no connection at all
+    if (ni == null || !ni.isConnected()) {
+      showSnackbar(R.string.ftp_server_prompt_connect_to_network, new Intent(ACTION_WIFI_SETTINGS));
+    } else {
+      // Data connection available, but no AP enabled
+      if (ni.getType() == ConnectivityManager.TYPE_MOBILE) {
+        showSnackbar(
+            R.string.ftp_server_prompt_open_ap, new Intent(Settings.ACTION_WIRELESS_SETTINGS));
```

Comment:
I think regardless of whether mobile data is enabled or not, we can always prompt user to open settings and enable wifi or start a hotspot.
Similarly, only one message should suffice 
`Connect to WiFi or enable hotspot`

Reply:
But what if the two snackbars would take user to different places in settings? As you can see here, `Open settings` would take user to wifi settings if device has no connection at all, or mobile network settings if user has mobile data enabled but didn't enable AP.

Reply:
Maybe just open him to enable wifi screen like you're doing now in case there's no network.
I mean, mobile network shouldn't matter because either way you have to enable wifi or hotspot.